vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx

```diff
@@ -370,16 +476,217 @@ module UnionTypes =
     /// This example prints all the cards in a playing deck.
     let printAllCards() = 
         for card in fullDeck do 
-            printfn "%s" (showCard card)
+            printfn "%s" (showPlayingCard card)
+
+    // Single-case DUs are often used for domain modeling.  This can buy you extra type safety
+    // over primitive types such as strings and ints.
+    type Address = Address of string
+    type Name = Name of string
+    type SSN = SSN of int
+
+    // You can easily instantiate a single-case DU as follows.
+    let address = Address "111 Alf Way"
+    let name = Name "Alf"
+    let ssn = SSN 1234567890
+
+    /// When you need the value, you can unwrap the underlying value with a simple function.
+    let unwrapAddress (Address a) = a
+    let unwrapName (Name n) = n
+    let unwrapSSN (SSN s) = s
+
+    // Printing single-case DUs is simple with unwrapping functions.
+    printfn "Address: %s, Name: %s, and SSN: %d" (address |> unwrapAddress) (name |> unwrapName) (ssn |> unwrapSSN)
+
+    /// Disciminated Unions also support recursive definitions.
+    ///
+    /// This represents a Binary Search Tree, with one case being the Empty tree,
+    /// and the other being a Node with a value and two subtrees.
+    type 'T BST =
+        | Empty
+        | Node of value:'T * left: BST<'T> * right: BST<'T>
+
+    /// Check if an item exists in the binary search tree.
+    /// Searches recursively using Pattern Matching.  Returns true if it exists; otherwise, false.
+    let rec exists item bst =
+        match bst with
+        | Empty -> false
+        | Node (x, left, right) ->
+            if item = x then true
+            elif item < x then (exists item left) // Check the left subtree.
+            else (exists item right) // Check the right subtree.
+
+    /// Inserts an item in the Binary Search Tree.
+    /// Finds the place to insert recursively using Pattern Matching, then inserts a new node.
+    /// If the item is already present, it does not insert anything.
+    let rec insert item bst =
+        match bst with
+        | Empty -> Node(item, Empty, Empty)
+        | Node(x, left, right) as node ->
+            if item = x then node // No need to insert, it already exists; return the node.
+            elif item < x then Node(x, insert item left, right) // Call into left subtree.
+            else Node(x, left, insert item right) // Call into right subtree.
+
+    /// Discriminated Unions can also be represented as structs via the 'Struct' attribute.
```

Comment:
Addressed this with a basic statement.  There should be some documentation somewhere where we could point to about structs v. reference types for performance, but since it's a bit of a tricky area that's subject to that specific code, I've opted not to explain anything here.